R/extract.R

```diff
@@ -1,11 +1,16 @@
 # content is the file content from readLines
-extract_r_source <- function(filename, lines) {
+extract_r_source <- function(filename, lines, error = identity) {
   pattern <- get_knitr_pattern(filename, lines)
   if (is.null(pattern$chunk.begin) || is.null(pattern$chunk.end)) {
     return(lines)
   }
 
-  chunks <- get_chunk_positions(pattern = pattern, lines = lines)
+  chunks <- tryCatch(get_chunk_positions(pattern = pattern, lines = lines), error = error)
+  if (!inherits(chunks, "list")) {
```

Comment:
why not `is.list` here?

Reply:
just for consistency with L246 (implementation of get_source_file).
WDYT about switching both to `inherits(., "error")` instead? That seems more clear to me.

Reply:
agree, if we can make the branch `if(x)` instead of `if(!y)` that's always better IMO